kodilla-exception/src/main/java/com/kodilla/exception/test/FlightFinder.java

```diff
@@ -0,0 +1,101 @@
+package com.kodilla.exception.test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FlightFinder {
+
+     public boolean findFlight(Flight flight) throws RouteNotFoundException{
+        Map<String, Boolean> avaibleAirport = new HashMap<>();
+
+         avaibleAirport.put("Warsaw", true);
+         avaibleAirport.put("Gdansk", true);
+         avaibleAirport.put("Vienna", true);
+         avaibleAirport.put("Prague", false);
+         avaibleAirport.put("Paris", false);
+         avaibleAirport.put("Berlin", false);
+         avaibleAirport.put("Hamburg", false);
+
+         String arrivalAirport = flight.getArrivalAirport();
+         String departureAirport = flight.getDepartureAirport();
+
+         Boolean isAvaibleArrivalAirpot = avaibleAirport.containsKey(arrivalAirport);
+         Boolean isAvaibleDepartureAirport = avaibleAirport.containsKey(departureAirport);
+
+         if (isAvaibleArrivalAirpot && isAvaibleDepartureAirport){
+
+              if(avaibleAirport.get(arrivalAirport) && avaibleAirport.get(departureAirport)){
+                  System.out.println("Flight avaible ");
+                  return true;
+              }
+              else if(!avaibleAirport.get(arrivalAirport)){
+                  System.out.println("No avaible flight on arrival airport");
+                  return false;
+              }
+              else{
+                  System.out.println("No avaible flight on departure airport");
+                  return false;
+              }
+         }
+         else {
+             String message;
+
+             if(!isAvaibleArrivalAirpot && !isAvaibleDepartureAirport){
+                 message = "No airport in any of those cities";
+             }
+             else if(!isAvaibleArrivalAirpot){
+                 message = "No airport in arrival location";
+
+             }
+             else{
+                 message = "No airport in departure location";
+             }
+
+             throw new RouteNotFoundException(message);
+         }
+    }
```

Comment:
powyzszy kod jest napisany w bardzo zawily sposob. Kazdy if sprawia ze 1) kod jest mniej czytelny 2) bedziesz musial napisac pozniej wiecej test caseow. Czy masz pomysl jak napisac to czytelniej ?

Reply:
Np. w RouteNotFoundException wystarczyloby zebys przekazal nazwe miasta (+ reszta messege'u zapisana w ciele RouteNotFoundException), bez if, else if itd